sklearn/metrics/tests/test_pairwise.py

```diff
@@ -1549,3 +1550,48 @@ def test_numeric_pairwise_distances_datatypes(metric, global_dtype, y_is_x):
     dist = pairwise_distances(X, Y, metric=metric, **params)
 
     assert_allclose(dist, expected_dist)
+
+
+def test_pairwise_dist_custom_scoring_for_string():
+    X = [
+        "a",
+        "ab",
+        "abc",
+    ]
+
+    def dummy_string_similarity(x, y):
+        return np.abs(len(x) - len(y))
+
+    actual_distance = pairwise_distances(
+        X, metric=dummy_string_similarity, check_length_only=True
+    )
+    expected_distance = np.array(
+        [
+            [0.0, 1.0, 2.0],
+            [1.0, 0.0, 1.0],
+            [2.0, 1.0, 0.0],
+        ]
+    )
+    assert_array_equal(actual_distance, expected_distance)
+
+    Y = ["a", "a", "a"]
+    actual_distance = pairwise_distances(
+        X, Y, metric=dummy_string_similarity, check_length_only=True
+    )
+
+    expected_distance = np.array(
+        [
+            [0.0, 0.0, 0.0],
+            [1.0, 1.0, 1.0],
+            [2.0, 2.0, 2.0],
+        ]
+    )
+    assert_array_equal(actual_distance, expected_distance)
+
+    Y.append("a")
+    msg = re.escape(
```

Comment:
You can isolate the test below in a test function since we are testing that it should raise an error.